### PR TITLE
Update react emotion

### PR DESCRIPTION
**Pull request by tshuck**
The package `react-emotion` has been [deprecated](https://www.npmjs.com/package/react-emotion) in favor of `@emotion/core` and `@emotion/styled`.  This can make it hard to look up current documentation on using the emotion library.  This PR:

  - Adds the new emotion packages
  - Changes the `styled` import to `@emotion/styled` in example templates
  - Removes `react-emotion`

I didn't see any tests, but running `npm start` shows the example site running as it was previously.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,8 @@
     "node": ">=8.0.0"
   },
   "dependencies": {
+    "@emotion/core": "^10.0.5",
+    "@emotion/styled": "^10.0.5",
     "babel-plugin-styled-components": "^1.7.1",
     "classnames": "^2.2.6",
     "emotion": "^9.2.9",
```
Comment (resir014) [inline package.json]: You can remove the old base `emotion` package as well.

Comment (resir014) [reply package.json]: Oh, and don't forget to upgrade `gatsby-plugin-emotion` to the latest version (`^4.0.1`) as well.

**Comment by tshuck** [comment]
@resir014 Removed `emotion` and `emotion-server` (which is [no longer needed with emotion 10](https://emotion.sh/docs/ssr)).  This required that we rework the `injectGlobal` (removed from emotion core) to the new `<Global>` component, as well as upgrade Gatsby due to [an issue with their plugin exports](https://github.com/gatsbyjs/gatsby/issues/10566)).

**Comment by resir014** [comment]
@tshuck Awesome. Lokks good to me. :+1:

Thanks for your help!

**Pull request merged**